src/ol/render/webgl/ShaderBuilder.js

```diff
@@ -844,7 +843,9 @@ ${this.attributes_
   float currentLengthPx = lengthToPoint + v_distanceOffsetPx;
   float currentRadiusPx = distanceFromSegment(currentPoint, v_segmentStart, v_segmentEnd);
   float currentRadiusRatio = dot(segmentNormal, startToPoint) * 2. / v_width;
-  currentLineMetric = mix(v_measureStart, v_measureEnd, lengthToPoint / segmentLength);
+  currentLineMetric = segmentLength == 0.0
+    ? v_measureStart
+    : mix(v_measureStart, v_measureEnd, lengthToPoint / segmentLength);
```

Comment:
Could you maybe use `max(segmentLength, epsilon)` instead of a ternary operator? Each branching in shader code can have impact on performance, and comparing with zero is not necessarily reliable either.

The `epsilon` value could be a very low value like `0.000001`.

Reply:
I changed it to use the minimum float value as a min value.

Only `0.0 / 0.0` results in NaN. The result of `1.0 / 0.0` result in Infinity, which would be handled correctly by the mix function.